### DISCUSSION TITLE
How can I build a Q&A agent on some data, where the code environment doesn't persist?

### DISCUSSION THREAD

**Discussion by abhipn** [microsoft/autogen#1606] [Q&A]
I'm attempting to construct a multi-agent system where I can load a CSV into memory using pandas, and then ask questions about the data for the agents to discuss and reach a solution. I want to deploy this Q&A system using Gradio, but when I transfer the logic to a .py file, it seems like the DataFrame isn't retained in memory. I could prompt it to read the file every time the code is executed, but that introduces additional latency due to the file's size. Is there a better way to accomplish this? In LangChain, I have utilized a pandas agent to achieve similar tasks, but I couldn't find anything helpful here for solving this problem

FYI: In a Jupyter notebook I was able to make it work.

**Comment by sonichi** [comment]
@ekzhu would you like to chime in?

**Comment by ekzhu** [comment]
Closing this one as it is duplicate with #1623. Replied in that thread.
